qiskit/chemistry/algorithms/pes_samplers/potentials/harmonic_potential.py

```diff
@@ -0,0 +1,266 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""This module implements a 1D Harmonic potential."""
+
+from typing import Tuple, List, Optional
+import numpy as np
+from scipy.optimize import curve_fit
+
+import qiskit.chemistry.constants as const
+from qiskit.chemistry.algorithms.pes_samplers.potentials.potential_base import PotentialBase
+from qiskit.chemistry.drivers import Molecule
+
+
+class HarmonicPotential(PotentialBase):
+    """
+    Implements a 1D Harmonic potential.
+    Input units are Angstroms (distance between the two atoms),
+        and output units are Hartrees (molecular energy).
+    """
+    # Works in Angstroms (input) and Hartrees (output)
+
+    def __init__(self, molecule: Molecule) -> None:
+        """
+        Args:
+            molecule: the underlying molecule.
+
+        Raises:
+            ValueError: Only implemented for diatomic molecules
+        """
+        # Initialize with zero-potential.
+        # Later - fit energy values (fit_to_data)
+        self.k = 0.0
+        self.m_shift = 0.0
+        self.r_0 = 0.0
+        self.d_e = None
+        self._m_a = molecule.masses[0]
+        self._m_b = molecule.masses[1]
```

Comment:
This class seems to have same problem as MorsePotential - I did not notice before

Reply:
This is now fixed. Thanks @stefan-woerner

Reply:
fixed